Failed to build log-source-app

#### Problem
Using instructions here to build log-source-app: https://github.com/project-chip/connectedhomeip/tree/master/examples/log-source-app/linux

Commit: 6f9fa8e80a9ada4ca7de38a0931c5b383658582a
Errors:
../../../examples/log-source-app/linux/main.cpp:71:53: error: format string is not a string literal [-Werror,-Wformat-nonliteral]
    chip::Logging::Platform::LogV(module, category, msg, argsCopy);
                                                    ^~~
../../../examples/log-source-app/linux/main.cpp:76:50: error: format string is not a string literal [-Werror,-Wformat-nonliteral]
    int s1 = vsnprintf(buffer1, sizeof(buffer1), msg, args);
                                                 ^~~
../../../examples/log-source-app/linux/main.cpp:78:83: error: implicit conversion changes signedness: 'int' to 'size_t' (aka 'unsigned long') [-Werror,-Wsign-conversion]
    GetLogProvider().PushLog(chip::ByteSpan(reinterpret_cast<uint8_t *>(buffer2), s2));

 #### Proposed Solution
log-source-app should build successfully

[build logs.txt](https://github.com/project-chip/connectedhomeip/files/8028742/build.logs.txt)

--- a/examples/log-source-app/linux/main.cpp
+++ b/examples/log-source-app/linux/main.cpp
@@ -63,7 +63,7 @@ DiagnosticLogsCommandHandler & GetLogProvider()
     return LogProvider;
 }
 
-void LoggingCallback(const char * module, uint8_t category, const char * msg, va_list args)
+void ENFORCE_FORMAT(3, 0) LoggingCallback(const char * module, uint8_t category, const char * msg, va_list args)
 {
     // Print the log on console for debug
     va_list argsCopy;
@@ -75,7 +75,9 @@ void LoggingCallback(const char * module, uint8_t category, const char * msg, va
     char buffer2[kMaxLogMessageLength];
     int s1 = vsnprintf(buffer1, sizeof(buffer1), msg, args);
     int s2 = snprintf(buffer2, sizeof(buffer2), "%s:%.*s", module, s1, buffer1);
-    GetLogProvider().PushLog(chip::ByteSpan(reinterpret_cast<uint8_t *>(buffer2), s2));
+
+    size_t len = chip::CanCastTo<size_t>(s2) ? static_cast<size_t>(s2) : SIZE_MAX;
+    GetLogProvider().PushLog(chip::ByteSpan(reinterpret_cast<uint8_t *>(buffer2), len));
 }
 
 int main(int argc, char * argv[])

